contracts/protocol/integration/SushiBarWrapAdapter.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.6.10;
+
+/**
+ * @title SushiBarWrapAdapter
+ * @author Yam Finance
+ *
+ * Wrap adapter for depositing/withdrawing sushi to/from SushiBar (xSushi)
+ */
+contract SushiBarWrapAdapter {
+
+    /* ============ State Variables ============ */
+
+
+    // Address of SUSHI token
+    address public immutable sushiToken;
+
+    // Address of xSUSHI token
+    address public immutable xsushiToken;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * State variables
+     *
+     * @param _sushiToken                      Address of SUSHI token
+     * @param _xsushiToken                     Address of XSUSHI token (also the SushiBar)
+     */
+    constructor(
+        address _sushiToken,
+        address _xsushiToken
+    )
+        public
+    {
+        sushiToken = _sushiToken;
+        xsushiToken = _xsushiToken;
+    }
+
+    /* ============ External Getter Functions ============ */
+
+    /**
+     * Generates the calldata to wrap sushi into xsushi.
+     *
+     * @param _underlyingToken      Address of the component to be wrapped
+     * @param _wrappedToken         Address of the wrapped component
+     * @param _underlyingUnits      Total quantity of underlying units to wrap
```

Comment:
Can maybe make note that this can only be SUSHI, same for wrappedToken and xSUSHI